test/proxy/Clones.t.sol

```diff
@@ -20,9 +20,11 @@ contract ClonesTest is Test {
         assertEq(spillage, bytes32(0));
     }
 
-    function testCloneDirty() external {
+    function testCloneDirty(address caller) external {
+        vm.startPrank(caller);
```

Comment:
This has no effect.

prank changes the address of `msg.sender` for outgoing calls. Here we are doing an internal library call and prank has no effect on it.

Reply:
Are you sure? The `clone` function uses `create` under the hood changing the resulting address.
I was thinking the test should prove that for any address, even if it's dirty, there's no side effect. I'm open to remove it.

Reply:
I'm sure of that part.

- `testCloneDirty` is a function that is executed bith some msg.sender
- doing `vm.startPrank(caller)` means that any outgoing call will see `caller` as their `msg.sender` (instead of `address(this)`
- then we do `Clone.clone` that is an internal function in a library, so basically its code is part of `address(this)`'s code, and it is executed in `address(this)` context.

Note that when a contract does a create, the caller (msg.sender) is irrelevant anyway.

Reply:
> Note that when a contract does a create, the caller (msg.sender) is irrelevant anyway.

Yes! Thanks for the clear explanation, you're right and this doesn't add any value as CREATE will result in the same address regardless